packages/providers/src/utils/change-case/functions.ts

```diff
@@ -0,0 +1,298 @@
+  return noCaseTransformer(input, { delimiter: '/', ...options });
+}
+
+/**
+ * Convert a string to path case (`Foo bar`).
+ */
+export function sentenceCaseTransformer(input: string, options?: IOptions) {
+  const [prefix, words, suffix] = splitPrefixSuffix(input, options);
+  const lower = lowerFactory(options?.locale);
+  const upper = upperFactory(options?.locale);
+  const transform = capitalCaseTransformFactory(lower, upper);
+
+  return (
+    prefix +
+    words
+      .map((word, index) => {
+        if (index === 0) return transform(word);
+
+        return lower(word);
+      })
+      .join(options?.delimiter ?? '') +
+    suffix
+  );
+}
+
+/**
+ * Convert a string to snake case (`foo_bar`).
+ */
+export function snakeCaseTransformer(input: string, options?: IOptions) {
+  return noCaseTransformer(input, { delimiter: '_', ...options });
+}
+
+/**
+ * Convert a string to header case (`Foo-Bar`).
+ */
+export function trainCaseTransformer(input: string, options?: IOptions) {
+  return capitalCaseTransformer(input, { delimiter: '-', ...options });
+}
+
+function lowerFactory(locale: Locale): (input: string) => string {
+  return locale === false
+    ? (input: string) => input.toLowerCase()
+    : (input: string) => input.toLocaleLowerCase(locale);
+}
+
+function upperFactory(locale: Locale): (input: string) => string {
+  return locale === false
+    ? (input: string) => input.toUpperCase()
+    : (input: string) => input.toLocaleUpperCase(locale);
+}
+
+function capitalCaseTransformFactory(
+  lower: (input: string) => string,
+  upper: (input: string) => string
+) {
+  return (word: string) => `${upper(word[0])}${lower(word.slice(1))}`;
+}
+
+function pascalCaseTransformFactory(
+  lower: (input: string) => string,
+  upper: (input: string) => string
+) {
+  return (word: string, index: number) => {
+    const char0 = word[0];
+    const initial =
+      index > 0 && char0 >= '0' && char0 <= '9' ? '_' + char0 : upper(char0);
+
+    return initial + lower(word.slice(1));
+  };
+}
+
+function splitPrefixSuffix(
+  input: string,
+  options: IOptions = {}
+): [string, string[], string] {
+  const splitFn =
+    options.split ?? (options.separateNumbers ? splitSeparateNumbers : split);
+  const prefixCharacters =
+    options.prefixCharacters ?? DEFAULT_PREFIX_SUFFIX_CHARACTERS;
+  const suffixCharacters =
+    options.suffixCharacters ?? DEFAULT_PREFIX_SUFFIX_CHARACTERS;
+  let prefixIndex = 0;
+  let suffixIndex = input.length;
+
+  while (prefixIndex < input.length) {
+    const char = input.charAt(prefixIndex);
+    if (!prefixCharacters.includes(char)) break;
+    prefixIndex++;
+  }
+
+  while (suffixIndex > prefixIndex) {
+    const index = suffixIndex - 1;
+    const char = input.charAt(index);
+    if (!suffixCharacters.includes(char)) break;
+    suffixIndex = index;
+  }
+
+  return [
+    input.slice(0, prefixIndex),
+    splitFn(input.slice(prefixIndex, suffixIndex)),
+    input.slice(suffixIndex),
+  ];
```

Comment:
<!-- Chores are simple tasks that must be done before the subject can be “officially” accepted. Usually, these comments reference some common process. Try to leave a link to the process description so that the reader knows how to resolve the chore.  -->
☑ **chore:** Rather than return a tuple which obfuscates the output variable intention, please return an object `{prefix, words, suffix}` which describes the interface fully for better encapsulation of logic (so that the consumer doesn't need to read through the function to understand what the outputs mean)‏